src/components/Devices/Devices.tsx

```diff
@@ -69,26 +65,17 @@ function Devices() {
   useEffect(() => {
     const getCollection = async () => {
       setIsLoaded(true);
-      const privateKey = window.localStorage.getItem(`temp_fastauthflow_${publicKeyFak}`);
-      if (privateKey) {
-        const accountId = await controller.getAccountIdFromOidcToken();
-
-        if (!window.fastAuthController) {
-          (window as any).fastAuthController = new FastAuthController({
-            accountId,
-            networkId
-          });
-        }
-        const keypair = new KeyPairEd25519(privateKey.split(':')[1]);
-        await window.fastAuthController.setKey(keypair);
-      }
-
       const deviceCollections = await controller.listDevices();
       setIsLoaded(false);
       setCollections(deviceCollections);
     };
-    if (authenticated !== 'loading') {
+    if (controller.getUserOidcToken()) {
       getCollection();
+    } else {
+      navigate({
+        pathname: '/login',
+        search:   searchParams.toString(),
+      });
```

Comment:
Have you encountered a situation where you had to redirect to login state?

Reply:
This should happen if you navigate to the devices page directly meaning currently the devices page requires an auth token and can't just be used with FAK. We should fix that soon